fk/luae_na/src/main/java/se/inera/intyg/common/luae_na/v1/model/converter/certificate/question/QuestionUnderlag.java

```diff
@@ -0,0 +1,206 @@
+                                .build(),
+                            MedicalInvestigation.builder()
+                                .typeId(UNDERLAG_SVAR_JSON_ID_4 + "[1].typ")
+                                .dateId(UNDERLAG_SVAR_JSON_ID_4 + "[1].datum")
+                                .informationSourceId(UNDERLAG_SVAR_JSON_ID_4 + "[1].hamtasFran")
+                                .build(),
+                            MedicalInvestigation.builder()
+                                .typeId(UNDERLAG_SVAR_JSON_ID_4 + "[2].typ")
+                                .dateId(UNDERLAG_SVAR_JSON_ID_4 + "[2].datum")
+                                .informationSourceId(UNDERLAG_SVAR_JSON_ID_4 + "[2].hamtasFran")
+                                .build()
+                        )
+                    )
+                    .build()
+            )
+            .value(
+                CertificateDataValueMedicalInvestigationList.builder()
+                    .list(List.of(
+                        getMedicalInvestigationValue(underlag, 0),
+                        getMedicalInvestigationValue(underlag, 1),
+                        getMedicalInvestigationValue(underlag, 2)
+                    ))
+                    .build()
+            )
+            .validation(
+                new CertificateDataValidation[]{
+                    CertificateDataValidationMandatory.builder()
+                        .questionId(UNDERLAG_TYP_DELSVAR_ID_4)
+                        .expression(multipleAndExpression(UNDERLAG_SVAR_JSON_ID_4 + "[0].typ", UNDERLAG_SVAR_JSON_ID_4 + "[0].datum",
+                            UNDERLAG_SVAR_JSON_ID_4 + "[0].hamtasFran"))
+                        .build(),
+                    CertificateDataValidationShow.builder()
+                        .questionId(UNDERLAGFINNS_DELSVAR_ID_3)
+                        .expression(singleExpression(UNDERLAGFINNS_SVAR_JSON_ID_3))
+                        .build(),
+                    CertificateDataValidationMaxDate.builder()
+                        .id(UNDERLAG_SVAR_JSON_ID_4 + "[0].datum")
+                        .numberOfDays(LIMIT)
+                        .build(),
+                    CertificateDataValidationMaxDate.builder()
+                        .id(UNDERLAG_SVAR_JSON_ID_4 + "[1].datum")
+                        .numberOfDays(LIMIT)
+                        .build(),
+                    CertificateDataValidationMaxDate.builder()
+                        .id(UNDERLAG_SVAR_JSON_ID_4 + "[2].datum")
+                        .numberOfDays(LIMIT)
+                        .build(),
+                }
+            )
+            .build();
+    }
+
+    private static CertificateDataValueMedicalInvestigation getMedicalInvestigationValue(List<Underlag> underlag, int id) {
+        return CertificateDataValueMedicalInvestigation.builder()
+            .typeId(UNDERLAG_SVAR_JSON_ID_4 + "[" + id + "].typ")
```

Comment:
Should we refactor this to make it more clear to why we set typeId here and below we also use a different typeId as a type?